### PR TITLE
Derive address scope instead of setting it explicitly in vsphere

### PR DESCRIPTION
**Pull request by voidspace**
Fixes bug #1629452
https://bugs.launchpad.net/juju/+bug/1629452 

Change the vsphere provider to use derived scope for all addresses instead of explicitly setting an incorrect public scope.

The vsphere provider has no infrastructure for configuring fake instances, so it's not possible to unit test this change. All existing unit tests pass though.

QA steps
Bootstrap to vsphere
Deploy Ubuntu units on xenial
Se that the units get IPv4 addresses and you can ssh to them.

### CODE REVIEW

File: state/machine.go [hunk 1]
```diff
@@ -1133,6 +1133,7 @@ func (m *Machine) SetInstanceInfo(
 	volumes map[names.VolumeTag]VolumeInfo,
 	volumeAttachments map[names.VolumeTag]VolumeAttachmentInfo,
 ) error {
+	logger.Tracef("Setting instance info: machine %v, deviceAddrs: %#v, devicesArgs: %#v", m.Id(), devicesAddrs, devicesArgs)
```
Comment (bz2) [inline state/machine.go]: Nit, are we starting log sentences with a capital or not?

File: provider/vsphere/instance.go [hunk 2]
```diff
@@ -53,11 +53,7 @@ func (inst *environInstance) Addresses() ([]network.Address, error) {
 	res := make([]network.Address, 0)
 	for _, net := range inst.base.Guest.Net {
 		for _, ip := range net.IpAddress {
-			scope := network.ScopeCloudLocal
-			if net.Network == inst.env.ecfg.externalNetwork() {
```
Comment (bz2) [inline provider/vsphere/instance.go]: Do we want to keep this logic still? If VMWare allows you to configure say, a 10. range as your external network (which makes sense for an internal setup perhaps), we'd currently be deriving that as CloudLocal when perhaps Public is more appropriate?

Comment (voidspace) [inline provider/vsphere/instance.go]: Well, it was this logic causing the specific bug that is fixed here...

Comment (bz2) [inline provider/vsphere/instance.go]: Okay, from discussing I agree trying to keep an override to public based on this config is mostly just harmful.

**Comment by voidspace** [comment]
$$merge$$

File: provider/vsphere/instance.go (hunk 2 above)
Comment (voidspace) [inline provider/vsphere/instance.go]: We can always add back in special casing if there is a need for it.

**Pull request merged**